dev-infrastructure/scripts/add-grafana-datasource.sh

```diff
@@ -27,21 +28,16 @@ EXISTING_DATA_SOURCE_URL=$(az grafana data-source list --name ${GRAFANA_NAME} \
 if [[ -n "${EXISTING_DATA_SOURCE_URL}" && ${EXISTING_DATA_SOURCE_URL} != ${PROM_QUERY_URL} ]];
 then
     echo "Removing ${MONITOR_NAME} integration from ${GRAFANA_NAME}"
-    az grafana integrations monitor delete \
-        --name ${GRAFANA_NAME} \
-        --resource-group ${GRAFANA_RG} \
-        --monitor-name ${MONITOR_NAME} \
-        --monitor-resource-group-name ${MONITOR_RG} \
-        --skip-role-assignment true
-    az resource wait --updated --ids $GRAFANA_RESOURCE_ID
+    MONITOR_UPDATES=$(echo "${MONITORS}" | jq --arg id "${MONITOR_ID}" 'map(select(.azureMonitorWorkspaceResourceId != $id))')#
```

Comment:
`az grafana integrations monitor delete` does not support cross-subscription cleanup, so have to go the manual `az resource update` road